### DISCUSSION TITLE
Is it possible to use this extension in drupal module packages?

### DISCUSSION THREAD

**Discussion by mad-briller** [mglaman/phpstan-drupal#467] [Q&A] [answered]
We've been using `phpstan-drupal` in our projects for a while now and it's been great, thank you for your work on it!

However we've tried using it when developing module packages and had little success, is it intended to be used in this context?  

It seems without setting a drupal root, the drupal-autoloader throws with `Unable to detect Drupal at`.
Is there a known solution to this problem that i'm missing? 

thanks for your time.

**Comment by mglaman** [comment, marked as answer]
@mad-briller is this on Drupal.org or your own continuous integration setup? How are you running your module's PHPUnit tests? That usually requires the module to be placed in a Drupal code base. You'd then add this to your Drupal site fixture.

I've built a GitHub Action for this: https://github.com/bluehorndigital/setup-drupal. It really just comes down to modifying however you setup the root Drupal site for testing.

**Reply by mad-briller** [reply]
thanks for the reference @mglaman, that helped a lot and we've managed to get it working

**Reply by mglaman** [reply]
That's great to hear!
